smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/auth/AuthUtils.java

```diff
@@ -5,25 +5,41 @@
 
 package software.amazon.smithy.typescript.codegen.auth;
 
-import java.util.function.Consumer;
+import java.util.Map;
+import java.util.TreeMap;
+import software.amazon.smithy.model.knowledge.ServiceIndex;
+import software.amazon.smithy.model.knowledge.ServiceIndex.AuthSchemeMode;
+import software.amazon.smithy.model.shapes.ServiceShape;
 import software.amazon.smithy.model.shapes.ShapeId;
-import software.amazon.smithy.typescript.codegen.TypeScriptWriter;
+import software.amazon.smithy.typescript.codegen.auth.http.HttpAuthScheme;
+import software.amazon.smithy.typescript.codegen.auth.http.SupportedHttpAuthSchemesIndex;
 import software.amazon.smithy.utils.SmithyInternalApi;
 
 /**
  * Auth utility methods needed across Java packages.
  */
 @SmithyInternalApi
 public final class AuthUtils {
-    /**
-     * Writes out `never`, which will make TypeScript compilation fail if used as a value.
-     */
-    public static final Consumer<TypeScriptWriter> DEFAULT_NEVER_WRITER = w -> w.write("never");
-
-    /**
-     * Should be replaced when the synthetic NoAuthTrait is released in Smithy.
-     */
-    public static final ShapeId NO_AUTH_ID = ShapeId.from("smithy.api#noAuth");
-
     private AuthUtils() {}
+
+    public static Map<ShapeId, HttpAuthScheme> getAllEffectiveNoAuthAwareAuthSchemes(
+        ServiceShape serviceShape,
+        ServiceIndex serviceIndex,
+        SupportedHttpAuthSchemesIndex authIndex
+    ) {
+        Map<ShapeId, HttpAuthScheme> effectiveAuthSchemes = new TreeMap<>();
```

Comment:
can you document why this map's order is important?

Reply:
For this method, the map order is actually not important. The `TreeMap` implementation was arbitrarily chosen.